Microarray ORA Restructure Instruction (#377)
AlexsLemonade/refinebio-examples: 02-microarray/pathway-analysis_microarray_01_ora.Rmd

This example is one of pathway analysis module set, we recommend looking at the [pathway analysis table below](#how-to-choose-a-pathway-analysis) to help you determine which pathway analysis method is best suited for your purposes.

> This particular example analysis shows how you can use over-representation analysis (ORA) to determine if a set of genes (e.g., those differentially expressed using some cutoff) shares more or fewer genes with gene sets/pathways than we would expect at random.

Reviewer: Just noticed this little wording change that seems minor but matters to me.
```suggestion
This particular example analysis shows how you can use over-representation analysis (ORA) to determine if a set of genes (e.g., those differentially expressed using some cutoff) shares more or fewer genes with gene sets/pathways than we would expect by chance.
```

# Using a different refine.bio dataset with this analysis?

> If you'd like to adapt an example analysis to use a different dataset from [refine.bio](https://www.refine.bio/), we recommend replacing the `dge_url` with a different file path to a gene list that you are interested in. 

Reviewer: We should specify what this file will contain: it is not just a gene list in this case, but a file with all the genes, each with p values and logFC. Not just any file will work in this code example.

I might add something about substituting the actual gene and background lists, rather than this specific file, with its specific format.

Author: Added some advice/caution about what needs to be changed.

> If you have differential expression results or something with a gene-level ranking and a two-group comparison, we recommend considering [GSEA](https://alexslemonade.github.io/refinebio-examples/02-microarray/pathway-analysis_microarray_02_gsea.html) for your pathway analysis questions. 

Reviewer: I think this sentence is likely to be missed here. I would move it to line 19, before you are getting into details about files and folders.

Author: Should I have it in both places?

Author: I left it in both places for now.

Reviewer: I don't think it actually makes much sense here. The file you are describing has gene level ranking, so this caution seems more likely to confuse than illuminate when put here.

Author: I removed it in this instance but not the other one that I added in line 19.

> As far as code, we still suggest saving plots and results to `plots/` and `results/` directories, respectively, as these are automatically created by the notebook.
> But, from here you can customize this ORA example to fit your own scientific questions and preferences.

Reviewer: Why did you change the wording here? I don't feel like we need to add the extra qualifications at this point: I'd go back to original wording: 
```suggestion
We suggest saving plots and results to `plots/` and `results/` directories, respectively, as these are automatically created by the notebook.
From here you can customize this analysis example to fit your own scientific questions and preferences.
```

## Import data

For ORA, we only need a list of gene IDs as our input, so this example can work for any situations where you have gene list and want to know more about what biological pathways it shares genes with.

> For this example, we will read in the differential expression results we will download from online.

Reviewer: ```suggestion
For this example, we will read in results from a differential expression analysis that we have already performed.
Rather than reading from a local file, we will download the results table directly from a URL.
```

These results are from a zebrafish microarray experiment we used for [differential expression analysis for two groups](https://alexslemonade.github.io/refinebio-examples/02-microarray/differential-expression_microarray_02_2-groups.html) using [`limma`](https://bioconductor.org/packages/release/bioc/html/limma.html) [@Ritchie2015].
The table contains Ensembl gene IDs, log fold-changes for each group, and adjusted p-values (FDR in this case).
We can identify differentially regulated genes by filtering these results and use this list as input to ORA.

> Generally, the dataset we are using here would be a good candidate for [GSEA](https://alexslemonade.github.io/refinebio-examples/02-microarray/pathway-analysis_microarray_02_gsea.html) since it has gene-level rankings, but here we are showing you how to use ORA since it can be useful in a number of other situations. 

Reviewer: I don't feel like we need to belabor this? If we want to say it again, I would probably put it toward the end with conclusions, rather than at this stage where we are focused on getting the data arranged for analysis.

Author: I deleted this mention

## Over-representation Analysis (ORA)

> Over-representation testing using `clusterProfiler` is based on a hypergeometric test [@clusterProfiler-book].

Reviewer: Unless there is objection, I think it is worth noting here that the hypergeometric test is equivalent to a Fisher's exact test, which people may be more familiar with. 
```suggestion
Over-representation testing using `clusterProfiler` is based on a hypergeometric test (often referred to as Fisher's exact test) [@clusterProfiler-book].
```
